202303-dplyr-joins/02-generate-monsoon-start.R

```diff
@@ -0,0 +1,41 @@
+# Obtain rainfall data near portal
+# Calculate monsoon start dates
+# Create table for non-equi joins
+library(tidyverse)
+
+# dat_url <- "https://zenodo.org/record/7735298/files/weecology/PortalData-4.0.0.zip"
+# dat_path <- "202303-dplyr-joins/data_raw/portaldata.zip"
+# 
+# download.file(dat_url, destfile = dat_path)
+# unzip(dat_path, exdir = "202303-dplyr-joins/data_raw")
+# 
+
+weather <- read_csv("202303-dplyr-joins/data_raw/weecology-PortalData-6017f65/Weather/Portal_weather.csv")
+# error can be ignored, not using `soiltemp`
+str(weather)
+
+#### Calculate monsoon start ####
+# sum to daily
+# calculate cumulative JAS
+mstart <- weather %>%
+  mutate(date = as.Date(timestamp)) %>%
+  group_by(date) %>%
+  summarize(year = unique(year),
+            month = unique(month),
+            ppt = sum(precipitation)) %>%
```

Comment:
This generates a warning in dplyr 1.1.0, but the suggested alternative, `reframe()`, doesn't exist prior to that version.  I'm guessing this is one of the things you've been thinking about in regards to whether to have people upgrade or not.  It's a good opportunity for a discussion about what "deprecated" means maybe.

It's possible that this is a bug though, since I can't think of why there wouldn't be one row per `date` in the output of this `summarize()`

Reply:
Or maybe this is all prep stuff and only the last script is for the actual workshop?

Reply:
Yeah, this is just prep stuff, won't be in the student version. I read over the summarize vs. reframe bit in the R4DS chapter but didn't fully grasp the difference